File: SpaceChallenge_Java/app/src/main/java/com/example/spacechallenge_java/U1.java
package com.example.spacechallenge_java;

public class U1 extends Rocket {

    public U1() {
        super(100, 10000, 18000, 0.05, 0.01);
        setTotalWeight(getSpaceshipWeight());
    }

    @Override
    public boolean launch() {
        double random = Math.random();
        return getChanceOfLaunchExplosion() * (getTotalWeight() - getSpaceshipWeight()) / (getMaxWeight() - getSpaceshipWeight()) <= random;
    }

    @Override
    public boolean land() {
        double random = Math.random();
        return getChanceOfLandingCrash() * (getTotalWeight() - getSpaceshipWeight()) / (getMaxWeight() - getSpaceshipWeight()) <= random;
    }
}

File: SpaceChallenge_Kotlin/app/src/main/java/com/savelievoleksandr/spacechallenge_kotlin/U1.kt
package com.savelievoleksandr.spacechallenge_kotlin

class U1 : Rocket(100, 10000, 18000, 0.05, 0.01) {

    override fun launch(): Boolean {
        val random = Math.random()
        return chanceOfLaunchExplosion * (totalWeight - spaceshipWeight) / (maxWeight - spaceshipWeight) <= random
    }

    override fun land(): Boolean {
        val random = Math.random()
        return chanceOfLandingCrash * (totalWeight - spaceshipWeight) / (maxWeight - spaceshipWeight) <= random
    }
}